main.go

```diff
@@ -83,16 +105,61 @@ func parseArgsAndLoadConfig(args []string) (conf configuration, listen string, a
 	}
 
 	confListen := strings.Split(conf.Server.Listen, ":")
-	if len(confListen) > 1 && string(port) != confListen[1] {
-		listen = fmt.Sprintf("%s:%d", confListen[0], port)
-		log.Infof("Overriding port from config %s with %d from the commandline", confListen[1], port)
+	if len(confListen) > 1 && port != confListen[1] {
+		listen = fmt.Sprintf("%s:%s", confListen[0], port)
+		log.Infof("Overriding listen addr from config %s with new port from the commandline: %s", conf.Server.Listen, listen)
 	} else {
 		listen = conf.Server.Listen
 	}
 
 	return
 }
 
+func startRuntimeCollector(conf configuration) {
+	if conf.Statsd.Addr == "" {
+		return
+	}
+
+	s, err := g2s.Dial("udp", conf.Statsd.Addr)
```

Comment:
I think using datadog's statsd is probably better. Then you can just write a wrapper that replaces the `g2s` functionality you're using atm.

If you do that, you can share a client too.